rajawali/src/main/java/org/rajawali3d/view/TextureView.kt

```diff
@@ -0,0 +1,1537 @@
+package org.rajawali3d.view
+
+import android.annotation.TargetApi
+import android.content.Context
+import android.graphics.SurfaceTexture
+import android.opengl.GLSurfaceView
+import android.os.Build
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import org.rajawali3d.R
+import org.rajawali3d.renderer.ISurfaceRenderer
+import org.rajawali3d.util.Capabilities
+import org.rajawali3d.util.egl.RajawaliEGLConfigChooser
+import java.lang.ref.WeakReference
+import java.util.*
+import javax.microedition.khronos.egl.*
+import javax.microedition.khronos.opengles.GL
+import javax.microedition.khronos.opengles.GL10
+
+/**
+ * Rajawali version of a [TextureView]. If you plan on using Rajawali with a [TextureView],
+ * it is imperative that you extend this class or life cycle events may not function as you expect.
+ *
+ * @author Jared Woolston (jwoolston@tenkiv.com)
+ */
+class TextureView : android.view.TextureView, ISurface {
```

Comment:
You can remove all the constructor declarations by using overloads

```kotlin
class TextureView @JvmOverloads constructor(
        context: Context,
        attributeSet: AttributeSet? = null,
        defStyleAttr: Int = 0
)
```

Then you can use init to run the apply call for the attrs